components/tidb_query_normal_expr/src/builtin_string.rs

```diff
@@ -1525,6 +1550,37 @@ mod tests {
         assert_eq!(got, expected, "repeat(NULL, NULL)");
     }
 
+    #[test]
+    fn test_insert() {
+        let cases = vec![
+            ("hello, world!", 1, 1, "asd", "asdello, world!"),
```

Comment:
Could you please add a case that the 3rd argument is 0, i.e. ("hello, world!", _, 0, _, _).